st2actions/st2actions/runners/httprunner.py

```diff
@@ -171,19 +173,36 @@ def run(self):
         results = {}
         resp = None
         try:
-            resp = requests.request(
-                self.method,
-                self.url,
-                params=self.params,
-                data=self.body,
-                headers=self.headers,
-                cookies=self.cookies,
-                auth=self.auth,
-                timeout=self.timeout,
-                allow_redirects=self.allow_redirects,
-                proxies=self.proxies,
-                files=self.files
-            )
+            if self._is_json_content():
+                # use the json property instead of data in case of json content
+                # also cast property to a dict.
+                resp = requests.request(
+                    self.method,
+                    self.url,
+                    params=self.params,
+                    json=self._cast_object(self.body),
```

Comment:
And to clarify it, this is mostly a style and not a logic change.

I do think though that wrapping `json.loads` and throwing a better exception on the error itself is worth it though - will make it easier to debug :)